agent/consul/fsm/commands_oss_test.go

```diff
@@ -1347,6 +1347,111 @@ func TestFSM_ConfigEntry(t *testing.T) {
 	}
 }
 
+func TestFSM_ConfigEntry_StatusCAS(t *testing.T) {
+	t.Parallel()
+
+	logger := testutil.Logger(t)
+	fsm, err := New(nil, logger)
+	require.NoError(t, err)
+
+	// Create a simple config entry
+	entry := &structs.APIGatewayConfigEntry{
+		Kind:           structs.APIGateway,
+		Name:           "global",
+		EnterpriseMeta: *structs.DefaultEnterpriseMetaInDefaultPartition(),
+		Status: structs.Status{
+			Conditions: []structs.Condition{{
+				Status: "Foo",
+			}},
+		}}
+
+	// Create a new request.
+	req := &structs.ConfigEntryRequest{
+		Op:    structs.ConfigEntryUpsertCAS,
+		Entry: entry,
+	}
+
+	{
+		buf, err := structs.Encode(structs.ConfigEntryRequestType, req)
+		require.NoError(t, err)
+		resp := fsm.Apply(makeLog(buf))
+		if _, ok := resp.(error); ok {
+			t.Fatalf("bad: %v", resp)
+		}
+	}
+
+	// Verify it's in the state store.
+	{
+		_, config, err := fsm.state.ConfigEntry(nil, structs.APIGateway, "global", nil)
+		require.NoError(t, err)
+		entry.RaftIndex.CreateIndex = 1
```

Comment:
what does the `1` here mean?

Reply:
It's the operation number in the raft log that's used as part of the CAS operation and resource version tracking. It gets autoincremented as ops are applied via a call to `fsm.Apply`. It ends up getting incremented here to make it match the entry retrieved from the store so that the subsequent CAS operations can succeed.